test/base/test_store.py

```diff
@@ -141,3 +145,83 @@ def test_storage_trace():
+    dim_full = 10
+    lb = -5 * np.ones((dim_full, 1))
+    ub = 5 * np.ones((dim_full, 1))
+    n_starts = 5
+    startpoints = pypesto.startpoint.latin_hypercube(n_starts=n_starts,
+                                                     lb=lb,
+                                                     ub=ub)
+    problem = pypesto.Problem(objective=objective, lb=lb, ub=ub,
+                              x_guesses=startpoints)
+
+    optimizer = pypesto.optimize.ScipyOptimizer()
+
+    result_optimization = pypesto.optimize.minimize(
+        problem=problem, optimizer=optimizer,
+        n_starts=n_starts)
+    profile_original = profile.parameter_profile(
+        problem=problem, result=result_optimization,
+        profile_index=[0], optimizer=optimizer)
+    with tempfile.TemporaryDirectory(dir=".") as tmpdirname:
+        _, fn = tempfile.mkstemp(".hdf5", dir=f"{tmpdirname}")
+
+    pypesto_profile_writer = ProfileResultHDF5Writer(fn)
+    pypesto_profile_writer.write(profile_original)
+    pypesto_profile_reader = ProfileResultHDF5Reader(fn)
+    profile_read = pypesto_profile_reader.read()
+
+    # compare the x_paths of both profiles
+    np.testing.assert_array_equal(
+        profile_original.profile_result.list[0][0]['x_path'],
+        profile_read.profile_result.list[0][0]['x_path']
+    )
+
+
+def test_storage_sampling():
+    objective = pypesto.Objective(fun=so.rosen,
+                                  grad=so.rosen_der,
+                                  hess=so.rosen_hess)
+    dim_full = 10
+    lb = -5 * np.ones((dim_full, 1))
+    ub = 5 * np.ones((dim_full, 1))
+    n_starts = 5
+    startpoints = pypesto.startpoint.latin_hypercube(n_starts=n_starts,
+                                                     lb=lb,
+                                                     ub=ub)
+    problem = pypesto.Problem(objective=objective, lb=lb, ub=ub,
+                              x_guesses=startpoints)
+
+    optimizer = pypesto.optimize.ScipyOptimizer()
+
+    result_optimization = pypesto.optimize.minimize(
+        problem=problem, optimizer=optimizer,
+        n_starts=n_starts)
+    x_0 = result_optimization.optimize_result.list[0]['x']
+    sampler = sample.AdaptiveParallelTemperingSampler(
+        internal_sampler=sample.AdaptiveMetropolisSampler(),
+        n_chains=1
+    )
+    sample_original = sample.sample(problem=problem,
+                                    sampler=sampler,
+                                    n_samples=100,
+                                    x0=[x_0])
+    with tempfile.TemporaryDirectory(dir=".") as tmpdirname:
+        _, fn = tempfile.mkstemp(".hdf5", dir=f"{tmpdirname}")
+
+    pypesto_sample_writer = SamplingResultHDF5Writer(fn)
+    pypesto_sample_writer.write(sample_original)
+    pypesto_sample_reader = SamplingResultHDF5Reader(fn)
+    sample_read = pypesto_sample_reader.read()
+
+    # compare the x_paths of both profiles
+    np.testing.assert_array_equal(
```

Comment:
same as above: check all content

Reply:
well all I think is not a good thing since time might vary. Will implement a check for all np.ndarrays, floats and ints aside from time